sympy/physics/quantum/tensorproduct.py

```diff
@@ -116,6 +117,17 @@ class TensorProduct(Expr):
     """
     is_commutative = False
 
+    @property
+    def is_Matrix(self):
+        return all(arg.is_Matrix for arg in self.args)
```

Comment:
I think this was discussed in some other issue, but it would be more useful to have a KroneckerProduct MatrixExpression class. That way you could have first-class support for operations with other matrices, and things like evaluation on explicit matrices.